src/pages/Category/CategoryList.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import GlobalStyles from '../../styles/GlobalStyles';
+
+function CategoryList(props) {
+  const [category, setCategory] = useState([]);
+  const [categoryList, setCategoryList] = useState([]);
+  const [categoryName, setCategoryName] = useState('');
+  const dropBar = React.createRef();
+
+  const fetchCategoryData = () => {
+    fetch('/data/categoryList.json')
+      .then(res => res.json())
+      .then(data => {
+        setCategory(data.results);
+      });
+  };
+
+  useEffect(() => {
+    fetchCategoryData();
+    window.addEventListener('click', removeDropBarMenu);
+  }, []);
+
+  const handleDropBarMenu = e => {
+    const target = e.target.closest('button');
+    const targetData = category[target.id - 1].categoryList;
+    const targetLeft = target.getBoundingClientRect().left;
+    setCategoryList(targetData);
+    dropBar.current.style.left = `${targetLeft}px`;
+  };
+
+  const removeDropBarMenu = e => {
+    if (!e.target.matches('li')) {
+      setCategoryList([]);
+    }
+  };
+
+  const getFilterName = e => {
+    let value = e.target.id;
+    const filterName = categoryList[value - 1].categoryName;
+    setCategoryName(filterName);
+    console.log(filterName);
+  };
```

Comment:
FilterName 설정하는 거는 굳이 어렵게 filterName을 찾아내는 로직을 넣는 것 보다 간단하게 표현 가능해보입니다!

```suggestion
  const getFilterName = categoryName => {
    setCategoryName(categoryName);
  };
```
```
                <DropBarList
                  onClick={()=>handleFilterBtn(el.categoryName}
                  key={el.categoryId}
                  id={el.categoryId}
                >
                  {el.categoryName}
                </DropBarList>
```

Reply:
시작한지 한달 조금 넘었는데 벌써 이만큼이나 하실 수 있게 되셨잖아요
앞으로도 이렇게 계속 공부하고 배워나가면 되죠!
같이 화이팅합시다!

Reply:
넹.... 나름 열심히 한다고 하는데.. 실력이 느는것 같지 않는건.. 기분 탓일까요? 팩트일까요..ㅎㅎㅎㅎㅎ 더더 열심히 하겠습니다 :) 화이팅!